bootloader/bootloader.go

```diff
@@ -146,17 +146,20 @@ type ExtractedRunKernelImageBootloader interface {
 // ManagedAssetsBootloader has its boot assets (typically boot config) managed
 // by snapd.
 type ManagedAssetsBootloader interface {
+	Bootloader
+
 	// IsCurrentlyManaged returns true when the on disk boot assets are managed.
 	IsCurrentlyManaged() (bool, error)
 	// ManagedAssets returns a list of boot assets managed by the bootloader
 	// in the boot filesystem.
 	ManagedAssets() []string
 	// UpdateBootConfig updates the boot config assets used by the bootloader.
 	UpdateBootConfig(*Options) error
-	// CommandLine returns the kernel command line composed of the built-in
-	// list and extra arguments passed in arguments. The command line may be
-	// different when using a bootloader in the recovery partition.
-	CommandLine(extra []string) (string, error)
+	// CommandLine returns the kernel command line composed of system run
+	// mode arguments, built-in bootloader specific static arguments
+	// followed by any extra arguments configured by snapd. The command line
+	// may be different when using a bootloader in the recovery partition.
```

Comment:
The name and the doc don't make clear that this will compute a command line based on what is on disk ATM (though not trusting it but for the edition). We'll need also something together with UpdateBootConfig to compute the command line of what is about to be written to disk. I don't want to block forward motion but this should maybe be called CurrentCommandLine and doc updated.

Reply:
I wasn't sure about the name change. CurrentCommandLine feels like it returns what the current command line is, i.e. the contents of /proc/cmdline, but in this case it's more like a projection of what the command shall be given the mode, system and extra parameters.

Reply:
So:

* the doc says built-in but that is indexed by what is on disk which the doc doesn't explain
* the interface names use "Currently" already to mean on disk,  so Current could mean from the point of view of the bootloader, not the running system

anyway just looking at this without predicting too much (I think for reseal will need somehow to hook into UpdateBootConfig which will complicate things and will need changes to both this and UpdateBootConfig):

* CurrentCommandLine would be fine
* Or ExpectedBootCommandLine  which means it's what we expect if we boot now
* or we have an argument that tells whether we are computing the command line for current state or for after an update,  checking whether one would happen,  which means either using the edition on disk, or the edition an update would write if one would occur